conan/tools/files/patches.py

```diff
@@ -118,3 +120,33 @@ def apply_conandata_patches(conanfile):
         else:
             raise ConanException("The 'conandata.yml' file needs a 'patch_file' or 'patch_string'"
                                  " entry for every patch to be applied")
+
+
+def export_conandata_patches(conanfile):
+    """
+    Exports patches stored in 'conanfile.conan_data' (read from 'conandata.yml' file). It will export
+    all the patches under 'patches' entry that matches the given 'conanfile.version'. If versions are
+    not defined in 'conandata.yml' it will export all the patches directly under 'patches' keyword.
+    """
+    if conanfile.conan_data is None:
+        raise ConanException("conandata.yml not defined")
+
+    patches = conanfile.conan_data.get('patches')
+    if patches is None:
+        conanfile.output.info("export_conandata_patches(): No patches defined in conandata")
+        return
+
+    if isinstance(patches, dict):
+        assert conanfile.version, "Can only be exported if conanfile.version is already defined"
+        entries = patches.get(conanfile.version, [])
+    elif isinstance(patches, list):
+        entries = patches
+    else:
+        raise ConanException("conandata.yml 'patches' should be a list or a dict {version: list}")
+    for it in entries:
+        patch_file = it.get("patch_file")
+        if patch_file:
+            src = os.path.join(conanfile.recipe_folder, patch_file)
+            dst = os.path.join(conanfile.export_sources_folder, patch_file)
+            mkdir(os.path.dirname(dst))
+            shutil.copy2(src, dst)
```

Comment:
if an element of the list doesn't contain a `patch_file` field, should it raise or display a warning?

Reply:
According to the logic, elements in the list can be string patches, if they define ``patch_string``, so it seems valid to skip it without warning or error

Reply:
Ok I see, so LGTM